src/store/actions/actionCreators.js

```diff
@@ -0,0 +1,20 @@
+import { SIGNUP } from './types';
+
+export const signUp = user => async dispatch => {
+  await fetch('https://authors-haven-odin.herokuapp.com/api/users/', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json; charset=utf-8',
+    },
+    body: JSON.stringify({ user }),
+  })
+    .then(res => res.json())
+    .then(data =>
+      dispatch({
+        type: SIGNUP,
+        payload: data,
+      }),
+    );
+};
+
```

Comment:
All actions that return a `function` instead of an object are trapped by the `thunk middleware` which removes the need for `async` and `await`. 

Secondly these functions are invoked  with an extra `http` argument which is basically a wrapper around fetch, (refer to `src/store/index.js`).

That means, if you wanted, you could simplify it like this
```js
export const signUp = user => (dispatch, getState, http) => {
  return http
    .post("users/", user)
    .then(data => dispatch({ type: SIGNUP, payload: data }));
};

```

Reply:
Okay noted thanks.  I will try to use this wrapper the next time. It is just that by the time I worked on this task the wrapper wasn't there.